lib/graphql/schema/field.rb

```diff
@@ -666,7 +666,12 @@ def resolve(object, args, query_ctx)
               inner_object = obj.object
 
               if defined?(@hash_key)
-                inner_object[@hash_key] || inner_object[@hash_key_str] || (@fallback_value != :not_given ? @fallback_value : nil)
+                hash_value = inner_object.to_h.key?(@hash_key) ? inner_object[@hash_key] : inner_object[@hash_key_str]
```

Comment:
Check if key exists first before trying to get the value. This is to cover the case when `inner_object[@hash_key] == false`.

Reply:
If a `OpenStruct` is provided instead of a `Hash` this is needed if one want to use `#key?` method on it. Will add a `#respond_to?` call before. That way no Hash copy should be created.